### ISSUE TITLE
Supporting extended resources

### ISSUE THREAD

**Issue by fabiand** [kubernetes-sigs/node-feature-discovery#229]
Similar to #45 this RFE is about enhancing NFD to also permit exposing extended resources.

Just like labels, extended resources can be used to expose compute related features of nodes, which do have some kind of limit.

The mechanics and permissions of how they get exposed are close to equal to labels.

**Comment by fabiand** [comment]
This was also raised here https://groups.google.com/forum/#!topic/kubernetes-sig-node/PDJ0wTTEdv4

**Comment by marquiz** [comment]
As I said in my reply to the email above, I'm favourable to the idea. This could fly,  expecting that there is a well justified use case, and, SIG-node is ok with this.

**Comment by zvonkok** [comment]
What specific extended resources is it all about? I do not think that NFD should advertise extended resources where one needs a driver deployed to detect the extended resource. 

I do not want to deploy a complete GPU stack to all nodes just to find out that 90% of my nodes are not GPU nodes. The pci label is for me the anchor to deploy all needed parts to have the card + stack activated and advertised to the cluster, which is of course not part of NFD.

Or is it here more about exposing the now labels as extended resources? 

Can someone shed some light, with a real-world example?

**Comment by mmirecki** [comment]
A real-world example:
Kubevirt needs to verify resources on a node to verify that a vm (wrapped in a pod) can be created on it.
This could for example be a bridge, ovs-bridge or vf to which a vm nic would be connected. For that we need to verify what resources are available on node.

**Comment by marquiz** [comment]
Do you need extended resources, more specifically, bookkeeping done by the master?

**Comment by mmirecki** [comment]
We need extended resources to track which nodes we can schedule our
workloads to.
Kubevirt provisions vm's inside pods. As an example, it's quite common for
a vm to have
more than one network, for which resources such as bridges, ovs-bridges or
virtual functions
are required. We need the extended resources to be able to schedule our
pod/vm's on
appropriate nodes.

**Comment by fabiand** [comment]
@zvonkok in our case no heavy logic is needed to gather the informationsin order to do the "labeling", we just need to express something else.

@marquiz the accounting is useful, as we don't consider some networking resources to be "indefinitely consumable", but are limited in fact. We'd like to use extended resources to reflect this fact.

**Comment by fabiand** [comment]
@mmirecki doy ou mind opening a PR to demonstrate how it could look?

**Comment by mmirecki** [comment]
Sure (have this waiting on my task queue for quite some time now).

**Comment by mmirecki** [comment]
Submitted PR adding bridge information:
https://github.com/kubernetes-sigs/node-feature-discovery/pull/252

**Comment by uniemimu** [comment]
I suggest to

/reopen
and
/remove-lifecycle rotten

as this feature would be really nice to have, we need it for specific hardware resources which are currently labeled via feature detection hooks. My experiments have shown that this is perfectly doable.

**Comment by marquiz** [comment]
Thanks @zvonkok, I was busy updating the readme...
